deeplenstronomy/utils.py

```diff
@@ -205,7 +207,12 @@ def draw_from_user_dist(filename, size, mode, step=10, df=None):
         
     else:
         raise NotImplementedError("unexpected mode passed, must be 'sample' or 'interpolate'")
-            
+
+    if isinstance(params, list):
+        params = [x for x in params if x != 'WEIGHT']
+        assert len(params) == len(parameters), "Parameter list wrong size!"
```

Comment:
Last change request.

The assert statement in utils.py goes against the style of the rest of the codebase. What I would prefer is for all validation lines to appear in check.py so that all checks happen before the simulation begins and there are no runtime errors that users have to deal with.

Could you please delete the assert statement here, and instead check that the parameter list is the correct length in check.py (in the spot where you were validating that it is a list)

Reply:
I can make this change, but it will require opening the userdist file during check.py before it's opened "for real" during the course of the actual code. If that's okay, I'll go ahead and make the changes, but I just wanted to check.

Reply:
Yup that is a-ok (and also how I other things in check.py), thanks for checking. After you've checked the column names are what you expect, insert a `del` statement to delete the userdist file from memory in check.py

Reply:
resolved by 26ef559